### ISSUE TITLE
Error when adding property to native prototypes

### ISSUE THREAD

**Issue by benjamingr** [babel/babel#2485]
> Plugin "builtin-pre" provided an invalid property of "foo".

And the REPL crashes. Reproduced 
![capture](https://cloud.githubusercontent.com/assets/1315533/10268501/47b13864-6ac3-11e5-92b5-125c851534dc.PNG)

**Comment by sebmck** [comment]
You're tainting the globals and adding an enumerable property. Why do you think this needs to be checked?

**Comment by benjamingr** [comment]
Hmm, let me rephrase. On my computer it freezes the REPL half the time and the other half throws that uncaught error. 

I don't think it should be checked in particular, I think that overall it's (ugly but) valid JavaScript and Babel shouldn't break on it.

**Comment by sebmck** [comment]
Well the entire environment is tainted when you're using the REPL. (Also note that as per the README, REPL and website related isuses should be in the [babel.github.io](https://github.com/babel/babel.github.io) which is why I'm still confused as to whether or not you think this should be fixed in the Babel compiler or the website/REPL.) This issue is probably relevant: babel/babel.github.io#161.

**Comment by benjamingr** [comment]
Well, the first thing I did was report the issue right away so it doesn't get "lost". I haven't isolated anything yet, I'll update the issue as I investigate it later I just figured that since most users don't report issues it would be preferable to report it partially than not at all. 

so - I haven't tested if it breaks on desktop too or not yet.

**Comment by sebmck** [comment]
I'm not sure what there is to isolate? The compiler is breaking because it's attempting to validate the plugins and it's enumerating over the `foo` property that was executed when you evaled your bit of code in the same environment.

**Comment by benjamingr** [comment]
I'm not sure I'm following you here, I don't mind reporting this in the website repo if it is preferable but this is definitely not correct behavior either on the REPL's side or the compiler itself.

**Comment by sebmck** [comment]
I don't see how this is an issue. A lot of libraries are going to break once you starting messing with globals.

**Comment by sebmck** [comment]
The bigger issue to me is that user code in the REPL is being evaluated in the same environment as the compiler, hence babel/babel.github.io#161.

**Comment by benjamingr** [comment]
> I don't see how this is an issue. A lot of libraries are going to break once you starting messing with globals.

Well, messing with globals is one thing and adding a property (that the library isn't adding itself of coursE) is another. I'm not saying it's good code but it definitely exists in the wild and Babel certainly doesn't _have_ to break it just for that sake. 

Although reading https://github.com/babel/babel.github.io/issues/161 this is clearly a REPL problem. It would not exhibit in other real world scenarios anyway since compilation isn't done on the fly and even if it is it's not in the same environment. 

Thanks for the fast response as always. I'll leave a comment there.
